pkg/internal/listergen/templates.go

```diff
@@ -0,0 +1,208 @@
+}
+
+{{ if  not .IsNamespaced -}}
+// Get retrieves the {{.Name}} from the indexer for a given name.
+func (s {{.NameLowerFirst}}Lister) Get(name string) (*{{$fqkind}}, error) {
+	key := apimachinerycache.ToClusterAwareKey(s.cluster.String(), "", name)
+	obj, exists, err := s.indexer.GetByKey(key)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errors.NewNotFound({{$typepkg}}.Resource("{{.NameLowerFirst}}"), name)
+	}
+	return obj.(*{{$fqkind}}), nil
+}
+{{ else -}}
+// {{.NameLowerFirst}} returns an object that can list and get {{.NameLowerFirst}}.
+func (s *{{.NameLowerFirst}}Lister) {{.Name}}s(namespace string) {{.Name}}NamespaceLister {
+	return {{.NameLowerFirst}}NamespaceLister{indexer: s.indexer, cluster: s.cluster, namespace: namespace}
+}
+
+// {{.Name}}NamespaceLister helps list and get {{.NameLowerFirst}}.
+// All objects returned here must be treated as read-only.
+type {{.Name}}NamespaceLister interface {
+	// List lists all {{.NameLowerFirst}} in the indexer for a given namespace.
+	// Objects returned here must be treated as read-only.
+	List(selector labels.Selector) (ret []*{{$fqkind}}, err error)
+	// Get retrieves the {{.Name}} from the indexer for a given namespace and name.
+	// Objects returned here must be treated as read-only.
+	Get(name string) (*{{$fqkind}}, error)
+	// Note(kcp): Workspace-capable Lister implementation doesn't support support expansions.
+	// {{.Name}}NamespaceListerExpansion
+}
+
+// {{.NameLowerFirst}}NamespaceLister implements the {{.Name}}NamespaceLister
+// interface.
+type {{.NameLowerFirst}}NamespaceLister struct {
+	indexer   cache.Indexer
+	cluster   logicalcluster.Name
+	namespace string
+}
+
+// List lists all {{.NameLowerFirst}} in the indexer for a given namespace.
+func (s {{.NameLowerFirst}}NamespaceLister) List(selector labels.Selector) (ret []*{{$fqkind}}, err error) {
+	selectAll := selector == nil || selector.Empty()
+
+	key := apimachinerycache.ToClusterAwareKey(s.cluster.String(), s.namespace, "")
+	list, err := s.indexer.ByIndex(apimachinerycache.ClusterAndNamespaceIndexName, key)
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range list {
+		obj := list[i].(*{{$fqkind}})
+		if selectAll {
+			ret = append(ret, obj)
+		} else {
+			if selector.Matches(labels.Set(obj.GetLabels())) {
+				ret = append(ret, obj)
+			}
+		}
+	}
+	return ret, err
+}
+
+// Get retrieves the {{.Name}} from the indexer for a given namespace and name.
+func (s {{.NameLowerFirst}}NamespaceLister) Get(name string) (*{{$fqkind}}, error) {
+	key := apimachinerycache.ToClusterAwareKey(s.cluster.String(), s.namespace, name)
+	obj, exists, err := s.indexer.GetByKey(key)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errors.NewNotFound({{$typepkg}}.Resource("{{.NameLowerFirst}}"), name)
+	}
+	return obj.(*{{$fqkind}}), nil
+}
+{{ end -}}
+`
+
+// Do we make this configurable?
+// TODO: reformat it to be able to configure Copyright Year
+const HeaderText = `
```

Comment:
This may not be needed, since we are taking the header text from an input flag. That logic should already be there in clientgen.